doc/user/content/sql/explain.md

```diff
@@ -25,244 +25,207 @@ into Materialize's behavior for specific queries, e.g. performance.
 
-`SQL -> raw plan -> decorrelated plan -> optimized plan -> dataflow`
+Explainee type | Description
+------|-----
+**select_stmt** | Display the plan for an ad-hoc select statement.
+**VIEW view_name** | Display the plan for an existing view.
+**MATERIALIZED VIEW view_name** | Display the plan for an existing materialized view.
 
-#### From SQL to raw plan
+### Output format
 
-In this stage, the planner:
+Users can select between `JSON` and `TEXT` output format.
 
-- Replaces SQL variable names with column numbers
-- Infers the type of each expression
-- Choose the correct overload for each function
+Output type | Description
+------|-----
+**TEXT** | The explanation output will be UTF-8 text.
+**JSON** | The explanation output will a JSON object.
 
-#### From raw plan to decorrelated plan
+### Explained stage
 
-In this stage, the planner:
+This determines the query optimization stage at which the plan snapshot will be taken.
+See the [Query Compilation Pipeline](#query-compilation-pipeline) section for details.
 
-- Replaces subqueries and lateral joins with non-nested operations
-- Replaces `OUTER` joins with lower-level operations
-- Replaces aggregate default values with lower-level operations
+Plan Stage | Description
+------|-----
+**RAW** | Display the raw plan.
+**DECORRELATED** | Display the decorrelated plan.
+**OPTIMIZED** | _(Default)_ Display the optimized plan.
+**PHYSICAL** | Display the physical plan.
 
-#### From decorrelated plan to optimized plan
+### Output modifiers
 
-In this stage, the planner:
+Output modifiers act as boolean toggles and can be combined in order to slightly tweak
+the information and rendering style of the generated explanation output.
 
-- Coalesces joins
-- Chooses join order and implementation
-- Fuses adjacent operations
-- Removes redundant operations
-- Evaluates any operations on constants
+Modifier | Description
+------|-----
+**arity** | Annotate each subplan with its number of produced columns. This is useful due to the use of offset-based column names.
+**join_impls** | Render details about the implementation strategy of physically optimized MIR `Join` nodes.
+**keys** | Annotate each subplan with its unique keys.
+**types** | Annotate each subplan with its inferred type.
 
-#### From optimized plan to dataflow
+## Query compilation pipeline
 
-In this stage, the planner:
+The job of the Materialize planner is to turn SQL code into a differential
+dataflow program. We get there via a series of progressively lower-level plans:
 
-- Maps plan operators to differential dataflow operators
-- Locates existing arrangements which can be reused
-- Installs the new dataflow into the running system
+```text
+SQL ⇒ raw plan ⇒ decorrelated plan ⇒ optimized plan ⇒ physical plan ⇒ dataflow
+```
 
-For more information on using `EXPLAIN`, see the [details](#details) below.
+#### From SQL to raw plan
 
-### Reading decorrelated/optimized plans
+In this stage, the planner:
```

Comment:
Maybe indicate somehow that this is not an exhaustive list of what optimizations happen at this stage. E.g.:
"In this stage, the planner performs various logical optimizations, for example:"